### DISCUSSION TITLE
Possibility to add url to sitemap with foreach

### DISCUSSION THREAD

**Discussion by denisp1993** [spatie/laravel-sitemap#366] [Q&A]
Hello, thanks for your amazing package.

I've some items paginated with vue js (load more way), so the crawler can't index this elements.
There's a way to do a foreach on model items like in this package?

https://github.com/dwightwatson/sitemap

**Example:**

`$posts = Post::all();

        foreach ($posts as $post) {
            Sitemap::addTag(route('posts.show', $post), $post->updated_at, 'daily', '0.8');
        }

        return Sitemap::render();`

I've try with this way, but the loop override my sitemap every time.

`$real_estate_objects = RealEstateObject::published()->get();

        foreach ($real_estate_objects as $key => $real_estate_object) {
            Sitemap::create()
                ->add(Url::create($real_estate_object->real_slug->it)
                    ->setLastModificationDate($real_estate_object->updated_at)
                    ->setChangeFrequency(Url::CHANGE_FREQUENCY_WEEKLY)
                    ->setPriority(0.1))
            ->writeToFile(public_path('sitemap-real-estate.xml'));
        }`

Thanks for any help.

**Comment by silicahd** [comment]
I ran into the same issue cant seem to find an answer for this.

**Comment by fgd007** [comment]
This is possible. Just unchain the code.

```
$sitemap = Sitemap::create();

foreach ($real_estate_objects as $key => $real_estate_object) {
    $sitemap->add(Url::create($real_estate_object->real_slug->it)
        ->setLastModificationDate($real_estate_object->updated_at)
        ->setChangeFrequency(Url::CHANGE_FREQUENCY_WEEKLY)
        ->setPriority(0.1))
}

$sitemap->writeToFile(public_path('sitemap-real-estate.xml'));
```
